Core/ComponentBaseClasses/elxTransformBase.hxx

```diff
@@ -653,6 +656,78 @@ TransformBase<TElastix>::WriteToFile(const ParametersType & param) const
     }
   }
 
+  const auto transformOutputFileNameExtensions =
+    this->m_Configuration->GetValuesOfParameter("TransformOutputFileNameExtensions");
+
+  if (!transformOutputFileNameExtensions.empty())
+  {
+    elxout << "WARNING: Support for the parameter TransformOutputFileNameExtensions is still experimental!\n"
+      "Transform files stored by this feature may still be incomplete or incorrect!"<< std::endl;  
+
+    // Initialize the factory.
+    itk::TransformFactoryBase::GetFactory();
+
+    const auto transformObject = [this]() -> itk::Object::ConstPointer {
+      const auto thisAsITKBase = this->GetAsITKBaseType();
+      assert(thisAsITKBase != nullptr);
+
+      // The string returned by elxGetClassName() corresponds to ITK's GetNameOfClass(),
+      // for AffineTransform, BSplineTransform, and TranslationTransform.
+      // Note that for EulerTransform and SimilarityTransform, ITK has "2D"
+      // or "3D" inserted in the class name.
```

Comment:
Assumes a correspondence between parameters of elastix transforms and ITK transforms, based on elastic class-name and the ITK name-of-class, for AffineTransform, BSplineTransform, TranslationTransform, EulerTransform (for 2D and 3D), SimilarityTransform (for 2D and 3D).